### ISSUE TITLE
[Bug]: Viewport cannot be changed

### ISSUE THREAD

**Issue by yogarasu** [storybookjs/storybook#26334]
### Describe the bug

@storybook/addon-viewport is broken.
If default viewport is set, you cannot change to a viewport other than defaultViewport on the ui.
Even if you change the viewport, it will return to the defaultviewport with a redirect-like behavior.
This doesn't happen on the latest 7.6.17, it happens on versions 8 and later (confirmed on 8.0.0-rc.2).

```preview.ts
import type { Preview } from "@storybook/react"
import { INITIAL_VIEWPORTS, MINIMAL_VIEWPORTS } from "@storybook/addon-viewport"

const preview: Preview = {
  parameters: {
    actions: { argTypesRegex: "^on[A-Z].*" }, // Delete this line after ver 8
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/i,
      },
    },
    viewport: {
      defaultViewport: "iphone5",
      viewports: {
        ...INITIAL_VIEWPORTS,
        ...MINIMAL_VIEWPORTS,
      },
    },
  },
}

export default preview
```

**Comment by vanessayuenn** [comment]
Thanks for reporting this issue! I can reproduce this this: https://stackblitz.com/edit/github-qjnbfp?file=.storybook%2Fpreview.ts

**Comment by shilman** [comment]
I've verified that this issue was introduced in `8.0.0-alpha.7` meaning that it's almost certainly due to https://github.com/storybookjs/storybook/pull/25423

**Comment by yogarasu** [comment]
@shilman 
I checked in my environment, and as you said, the problem occurs from 8.0.0-alpha.7.

**Comment by yogarasu** [comment]
@shilman 
Thank you!!!
I was able to confirm that the behavior is fixed in 8.0.0-rc.3.
